kythe/cxx/indexer/cxx/IndexerASTHooks.cc

```diff
@@ -3827,856 +3992,606 @@ void IndexerASTVisitor::DumpTypeContext(unsigned Depth, unsigned Index) {
   }
 }
 
-absl::optional<GraphObserver::NodeId> IndexerASTVisitor::BuildNodeIdForType(
-    const clang::TypeLoc& Type, EmitRanges EmitRanges) {
-  return BuildNodeIdForType(Type, Type.getTypePtr(), EmitRanges);
+NodeSet IndexerASTVisitor::BuildNodeSetForBuiltin(
+    clang::BuiltinTypeLoc TL) const {
+  return Observer.getNodeIdForBuiltinType(TL.getTypePtr()->getName(
+      clang::PrintingPolicy(*Observer.getLangOptions())));
 }
 
-absl::optional<GraphObserver::NodeId> IndexerASTVisitor::BuildNodeIdForType(
-    const clang::TypeLoc& Type, const clang::QualType& QT,
-    EmitRanges EmitRanges) {
-  return BuildNodeIdForType(Type, QT.getTypePtr(), EmitRanges);
+NodeSet IndexerASTVisitor::BuildNodeSetForEnum(clang::EnumTypeLoc TL) {
+  EnumDecl* Decl = TL.getDecl();
+  if (EnumDecl* Defn = Decl->getDefinition()) {
+    return {BuildNodeIdForDecl(Defn), GraphObserver::Claimability::Unclaimable};
+  }
+  // If there is no visible definition, Id will be a tnominal node
+  // whereas it is more useful to decorate the span as a reference
+  // to the visible declaration.
+  // See https://phabricator-dot-kythe-repo.appspot.com/D1887
+  return {BuildNominalNodeIdForDecl(Decl), BuildNodeIdForDecl(Decl)};
 }
 
-absl::optional<GraphObserver::NodeId> IndexerASTVisitor::BuildNodeIdForType(
-    const clang::QualType& QT) {
-  CHECK(!QT.isNull());
-  TypeSourceInfo* TSI = Context.getTrivialTypeSourceInfo(QT, SourceLocation());
-  return BuildNodeIdForType(TSI->getTypeLoc(), EmitRanges::No);
+NodeSet IndexerASTVisitor::BuildNodeSetForRecord(clang::RecordTypeLoc TL) {
+  // TODO(shahms): Caching.
```

Comment:
nit: cache what and why

Reply:
Thanks, obsolete (caching is handled by `BuildNodeSetForType`)